axelor-account/src/main/java/com/axelor/apps/account/web/InvoicePaymentController.java

```diff
@@ -193,4 +196,88 @@ public void checkConditionBeforeSave(ActionRequest request, ActionResponse respo
       TraceBackService.trace(response, e, ResponseMessageType.ERROR);
     }
   }
+
+  public void computeDatasForFinancialDiscount(ActionRequest request, ActionResponse response) {
+    try {
+      InvoicePayment invoicePayment = request.getContext().asType(InvoicePayment.class);
+      Long invoiceId =
+          Long.valueOf(
+              (Integer) ((LinkedHashMap<?, ?>) request.getContext().get("_invoice")).get("id"));
+      if (invoiceId > 0) {
+        Invoice invoice = Beans.get(InvoiceRepository.class).find(invoiceId);
+        Boolean applyDiscount = Beans.get(InvoiceService.class).applyFinancialDiscount(invoice);
+        response.setValue("applyFinancialDiscount", applyDiscount);
+        response.setValue(
+            "amount",
+            Beans.get(InvoiceService.class)
```

Comment:
Call `Beans.get()` only once and store the service in a variable to avoid performance issues.